src/Presentation/Rest/GetMappingsApi.php

```diff
@@ -0,0 +1,61 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace ProfessionalWiki\WikibaseRDF\Presentation\Rest;
+
+use MediaWiki\Rest\SimpleHandler;
+use ProfessionalWiki\WikibaseRDF\Application\MappingRepository;
+use ProfessionalWiki\WikibaseRDF\WikibaseRdfExtension;
+use RequestContext;
+use Wikibase\DataModel\Entity\EntityId;
+use Wikibase\DataModel\Entity\EntityIdParser;
+use Wikimedia\ParamValidator\ParamValidator;
+
+class GetMappingsApi extends SimpleHandler {
+
+	public static function factory(): self {
+		return new self(
+			WikibaseRdfExtension::getInstance()->newEntityIdParser(),
+			WikibaseRdfExtension::getInstance()->newMappingRepository(
+				RequestContext::getMain()->getUser()
+			)
+		);
+	}
```

Comment:
I did consider that. However, since the factory method has to be static, getting the instance of the top level class inside a static method on that class just looked a bit odd:
```php
	public static function newGetMappingsApi(): GetMappingsApi {
		return new GetMappingsApi(
			self::getInstance()->newEntityIdParser(),
			self::getInstance()->newMappingRepository( RequestContext::getMain()->getUser() )
		);
	}
```
Or maybe that's fine and it's better to have all the factory stuff in there?

Reply:
What about keeping the static function in the API class and having an instance method in our factory? Then we have the dependency graph construction in our factory while dealing with the static requirement. Or could have both an instance and a differently named static method in our factory.